colin/checks/abstract/dockerfile.py

```diff
@@ -14,28 +14,64 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 import logging
+import re
 
-from .labels import check_label
+from ..check_utils import check_label
 from ..result import CheckResult
 from .abstract_check import AbstractCheck
 
 logger = logging.getLogger(__name__)
 
 
+def get_instructions_from_dockerfile_parse(dfp, instruction):
+    """
+    Get the list of instruction dictionary for given instruction name.
+    (Subset of DockerfileParser.structure only for given instruction.)
+
+    :param dfp: DockerfileParser
+    :param instruction: str
+    :return: list
+    """
+    result = []
+    for inst in dfp.structure:
+        if inst["instruction"] == instruction:
+            result.append(inst)
+    return result
```

Comment:
I know you know list comprehensions, I'm just wondering why you don't use it here:

```python
return [inst for inst in dfp.structure if inst["instruction"] == instruction]
```

Reply:
Neither am I...;-) Thanks!